### DISCUSSION TITLE
Typescript declaration files

### DISCUSSION THREAD

**Discussion by Mister-Hope** [walinejs/waline#213] [Q&A] [answered]
I am the author of [vuepress-theme-hope](https://github.com/vuepress-theme-hope/vuepress-theme-hope) and I just know waline from https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/602.

Currently I have plan to support valine in my theme. But my projects are all written by TypeScript. While waline is written by JavaScript. My questions are:

- Does Waline have plans to migrate to TypeScript?
- If above is 'No', does Waline have plans to add declaration files for `@waline/client`?
- If above is 'No', would Waline accept declaration files PR for `@waline/client` if someone sent one? (I can make the PR when I am adding support in my theme)

Thanks.

**Comment by lizheming** [comment, marked as answer]
I'm currently trying to rewrite the client using Svelte, but the release time is uncertain. I think the easy way is to add a declaration file, it also can be used in valine. And I also welcome someone to help implement it.
